### PR TITLE
Profile refresh & resolution

### PR DESCRIPTION
**Pull request by bfollington**
Fixes https://github.com/subconsciousnetwork/subconscious/issues/684
Fixes https://github.com/subconsciousnetwork/subconscious/issues/699

# Changes

- Changed "unresolved" icon to be less prominent (screenshot below)
- Introduced "Refresh" option in context menu
- Sync with gateway when doing pull-to-refresh on profile
- Slightly increase wait duration for resolution

image

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/Detail/UserProfileDetailView.swift [hunk 1]
```diff
@@ -53,7 +53,10 @@ struct UserProfileDetailView: View {
             onNavigateToNote: self.onNavigateToNote,
             onNavigateToUser: self.onNavigateToUser,
             onProfileAction: self.onProfileAction,
-            onRefresh: store.refresh
+            onRefresh: {
+                app.send(.syncAll)
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Components/Detail/UserProfileDetailView.swift:57]: 👍

File: xcode/Subconscious/Shared/Services/AddressBookService.swift [hunk 2]
```diff
@@ -355,7 +355,7 @@ actor AddressBookService {
     func waitForPetnameResolution(
         petname: Petname
     ) async throws -> Cid? {
-        let maxAttempts = 10 // 1+2+4+8+16+32+32+32+32+32 = 191 seconds
+        let maxAttempts = 13 // 1+2+4+8+16+32+32+32+32+32+32+32+32 = 287 seconds
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Services/AddressBookService.swift:358]: 🙏

File: xcode/Subconscious/Shared/Components/Common/Story/StoryUserView.swift [hunk 3]
```diff
@@ -105,7 +106,23 @@ struct StoryUserView: View {
...
+                        
+                        switch story.user.resolutionStatus {
+                        case .unresolved:
+                            Button(
+                                action: {
+                                    onRefreshUser()
+                                },
+                                label: {
+                                    Label(
+                                        title: { Text("Refresh") },
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Components/Common/Story/StoryUserView.swift:118]: My gut is that an explicit refresh control feels like exposing too much implementation detail to the user.

Could it be possible to invoke this automatically for any unresolved profile, or tie it into following list pull-to-refresh?

Non-blocking. We can file a follow-up to address.

Comment (bfollington) [reply xcode/Subconscious/Shared/Components/Common/Story/StoryUserView.swift:118]: Tying it to pull-to-refresh is a good idea, I'll experiment with that before merging to investigate.

Comment (bfollington) [reply xcode/Subconscious/Shared/Components/Common/Story/StoryUserView.swift]: As discussed, attempting resolution again on-tap feels way better.


https://github.com/subconsciousnetwork/subconscious/assets/5009316/f4e69a8d-d6f5-4e28-8a3e-b9dea34884cd

**Pull request merged**